designs/component-upgrades.md

```diff
@@ -0,0 +1,149 @@
+# EKS-A components upgrade
+
+
+## Introduction
+
+**Problem:** customers are only able to upgrade the cluster components that are part of the kubernetes distribution. Any other core component of EKS-A remains untouched from the cluster creation.
+
+This stops customers from enjoying the latest features and bug fixes from eks-a and any of its core components as well as from being able to get security patches.
+
+### Tenets
+
+* **Simple:** simple to use, simple to maintain, simple to build
+
+### Goals and objectives
+
+As an eks-a customer:
+
+* I want to able to enjoy the latest eks-a features and bug fixes
+* I want to keep my cluster up to date with security patches
+* I don’t want to need to stay informed about latest versions, security patches or compatibility between components
+
+### Statement of scope
+
+**In scope**
+
+* Upgrade core eks-a components to latest versions available for an specific cli version
+    * Core CAPI
+    * CAPI providers
+    * Etcdadm CAPI provider
+    * CNIs (cilium)
+    * EKS-A controller and CRDs
+    * Flux
```

Comment:
Should we also include cert-manager in this list?

Reply:
Good idea
I will also mention it in the CAPI upgrade section, since `clusterctl` will update it as well